recipes/hwloc/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+
+    def layout(self):
+        if self.settings.os == "Windows":
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            tc = CMakeToolchain(self)
+            tc.cache_variables["HWLOC_ENABLE_TESTING"] = 'OFF'
+            tc.cache_variables["HWLOC_SKIP_LSTOPO"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_TOOLS"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_INCLUDES"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_OPENCL"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_CUDA"] = 'OFF'
+            tc.cache_variables["HWLOC_BUILD_SHARED_LIBS"] = self.options.shared
+            tc.cache_variables["HWLOC_WITH_LIBXML2"] = not self.options.disable_libxml2
+            tc.generate()
+        else:
+            deps = AutotoolsDeps(self)
+            deps.generate()
+            tc = AutotoolsToolchain(self)
+            if self.options.disable_libxml2:
+                tc.configure_args.extend(["--disable-libxml2"])
+            tc.configure_args.extend(["--disable-io", "--disable-cairo"])
+            tc.generate()
+
+    def build(self):
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.configure(build_script_folder=os.path.join("contrib", "windows-cmake"))
+            cmake.build()
+        else:
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
+
+    def package(self):
+        copy(self, "COPYING", self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.install()
+            # remove PDB files
+            rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        else:
+            autotools = Autotools(self)
+            autotools.install()
+            # remove tools
+            rmdir(self, os.path.join(self.package_folder, "bin"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_module_file_name", "hwloc")
+        self.cpp_info.set_property("cmake_file_name", "hwloc")
+        self.cpp_info.set_property("cmake_target_name", "hwloc::hwloc")
+        self.cpp_info.set_property("pkg_config_name", "hwloc")
+        self.cpp_info.libs = ["hwloc"]
+
+        if not self.options.shared:
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs = ["m"]
+            if not self.options.disable_libxml2:
+                self.cpp_info.requires = ["libxml2::libxml2"]
```

Comment:
isn't this automatically propagated from self.requires call if we don't use components?

Reply:
good point, removed.